types/ag-auth/index.d.ts

```diff
@@ -0,0 +1,23 @@
+// Type definitions for ag-auth 1.0
+// Project: https://github.com/SocketCluster/ag-auth
+// Definitions by: Daniel Rose <https://github.com/DanielRose>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 3.7
+
+import { VerifyOptions, Secret, SignOptions, Jwt } from 'jsonwebtoken';
+
+declare class AuthEngine implements AuthEngine.AGAuthEngine {
+    constructor();
+
+    verifyToken(signedToken: string | null, key: Secret, options?: VerifyOptions): Promise<Jwt>;
+    signToken(token: string | object | Buffer, key: Secret, options?: SignOptions): Promise<string | undefined>;
+}
+
+export = AuthEngine;
+
+declare namespace AuthEngine {
```

Comment:
why do you need those? The package just declares a class like constructor and exports it IMO:
https://github.com/SocketCluster/ag-auth/blob/master/index.js#L6

Reply:
One can have this 'AuthEngine' contract within declarin options module (socker server). In TS the class here can be just same shape, but wihtout hard contract between modules